examples/app-cpp/payload_app.cc

```diff
@@ -395,6 +399,41 @@ void wakeup_seq_fsm(mythreadState_t *threadState)
     pthread_cond_signal(&threadState->condition);
 }
 
+AntarisReturnCode process_req_payload_metrics(ReqPayloadMetricsParams *payload_metrics_param)
+{
+    PayloadMetricsResponse   resp_payload_metrics_params;
+
+    printf("Before resp_payload_metrics_params\n");
+
+    time_t now = time(0);
+    UINT64 epoch = static_cast<UINT64>(now);
+    resp_payload_metrics_params.correlation_id = payload_metrics_param->correlation_id;
+    resp_payload_metrics_params.used_counter = 8;   // Example value
+    resp_payload_metrics_params.timestamp = epoch;
+
+    // Total number of counters supported is 8, hence setting all names to 0
+    for (int i = 0; i < USED_COUNTER_MAX ; i++) {
+        resp_payload_metrics_params.metrics[i].counter = 0; 
```

Comment:
What if we change array size in xml file? Do I need to change here as well?

can u use something like:

    size_t metrics_size = sizeof(((struct PayloadMetricsResponse*)0)->metrics) / sizeof(struct PayloadMetricsInfo);

Reply:
Change Done. XML changes will be automatically reflected here.